docs/userguide/core-patterns.md

```diff
@@ -1,3 +1,70 @@
-# Patterns
+# Workflow Core: Patterns/Variations
 
-_Coming soon!_
+There are a lot associated/generic types in workflow code – that doesn't mean you always need to use
+all of them. Here are some common configurations we've seen.
+
+## Stateless Workflows
+
+Remember that workflow state is made up of public and private parts. When a workflow's state
+consists entirely of public state, it can ignore all the machinery for private state. In Swift, the
+`State` type can be `Void`, and in `Kotlin` it can be `Unit` – such workflows are often referred to
+as "stateless", since they have no state of their own.
+
+## Inputless Workflows
+
+Some workflows manage all of their state internally, and have no public state (aka input). In Swift,
+this just means the workflow implementation has no parameters (although this is rare, see
+_Injecting Dependencies_ below). In Kotlin, the `Input` type can be `Unit`. `RenderContext` has
+convenience overloads of most of its functions to implicitly pass `Unit` for these workflows.
+
+## Outputless Workflows
+
+Workflows that only talk to their parent via their `Rendering`, and never emit any output, are
+encouraged to indicate that by using the [bottom type](https://en.wikipedia.org/wiki/Bottom_type) as
+their `Output` type. In addition to documenting the fact that the workflow will never output, using
+the bottom type also lets the compiler enforce it – code that tries to emit outputs will not
+compile. In Swift, the `Output` type is specified as [`Never`](https://nshipster.com/never/). In
+Kotlin, use [`Nothing`](https://medium.com/@agrawalsuneet/the-nothing-type-kotlin-2e7df43b0111).
+
+## Composite Workflows
+
+Composition is a powerful tool for working with Workflows. A workflow can often accomplish a lot
+simply by rendering various children. It may just combine the renderings of multiple children, or
+use its input to determine which of a set of children to render. Such workflows can often be
+stateless.
+
+## Input values v. Injected Dependencies
+
+[Dependency injection](https://en.wikipedia.org/wiki/Dependency_injection) is a technique for making
+code less coupled and more testable. In short, it's better for classes/structs to accept their
+dependencies when they're created instead of hard-coding them. Workflows typically have dependencies
+like specific Workers they need to perform some tasks, child workflows to delegate rendering to, or
+helpers for things like network requests, formatting and logging.
+
+### Swift
+
+A Swift workerflow typically receives its dependencies as initializer arguments, just like its input
```

Comment:
`workerflow` -> `workflow`

Reply:
**workerflow** – a workflow that just wraps a worker? 😅